userspace/libsinsp/filterchecks.h

```diff
@@ -164,7 +170,13 @@ class sinsp_filter_check : public gen_event_filter_check
 	check_extraction_cache_entry* m_extraction_cache_entry = NULL;
 
 protected:
+	// This is a single-value version of extract for subclasses non supporting extracting
+	// multiple values. By default, this returns NULL.
+	// Subclasses are meant to either override this, or the multi-valued extract method.
+	virtual uint8_t* extract(sinsp_evt *evt, OUT uint32_t* len, bool sanitize_strings = true);
```

Comment:
What about rename this to `extract_single_val` or something similar?

Reply:
I chose to leave the method name as is to let sinsp filtercheck keep using it, without the need or changing the name in a lot of code parts. I would agree with this, but we may expect a much larger PR if this is the case.

However, I made it protected as a defence measure. This way, the method can only be used internally for filterchecks not supporting multi-value extraction, but it is not exposed to external consumers.

Reply:
> However, I made it protected as a defence measure. This way, the method can only be used internally for filterchecks not supporting multi-value extraction, but it is not exposed to external consumers.

Yep, i don't really like that from a quick read it is unpredictable when we use one or the other (ok they take different parameters, but still it's confusing me a bit).
But i am okay with that, given that we want to avoid a huge diff.

Reply:
If we end up merging this one, later we can make a clean-up PR just for this!